script.go

```diff
@@ -898,6 +911,14 @@ func (p *Pipe) WithStdout(w io.Writer) *Pipe {
 	return p
 }
 
+// WithEnv sets the pipe's environment to the string array env. This will override
```

Comment:
One small suggestion, which relates to the discussion on the main page too, is that this doesn't modify the "pipe" environment, but the "next exec command's" environment. If thie documentation mentions this I believe it's clear and won't be confusing to folks accessing the env outside of the `Exec` functions.

Reply:
@lambrospetrou thanks for the suggestion! i clarified this comment now to add more detail about the usage. let me know what you think and if you have any other feedback